### PR TITLE
Extract Fuser from proc/fuse.Proc

### PR DESCRIPTION
**Pull request by nwt**
Groundwork for #1271.

### CODE REVIEW

File: zng/fuser/fuser.go [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+)
+
+// Fuser buffers records written to it, assembling from them a unified schema of
+// fields and types.  Fuser then transforms those records to the unified schema
+// as they are read back from it.
+type Fuser struct {
+	zctx        *resolver.Context
+	memMaxBytes int
+
+	spiller  *spill.File
```
Comment (alfred-landrum) [inline zng/fuser/fuser.go:20]: nit: sort this & below field names.

File: proc/fuse/fuse.go [hunk 2]
```diff
@@ -18,22 +15,17 @@ var BatchSize = 100
...
-	spiller  *spill.File
-	slotByID [][]int
-	slots    []slot
-	uberType *zng.TypeRecord
 	once     sync.Once
 	resultCh chan proc.Result
-	nbytes   int
-	recs     []*zng.Record
-	types    resolver.Slice
+	fuser    *fuser.Fuser
```
Comment (alfred-landrum) [inline proc/fuse/fuse.go:20]: nit: move above `once`.

**Review by alfred-landrum — COMMENTED** [review]
The test failure looks this existing fuse issue: #1702 , not necessarily something in this PR's changes.

File: proc/fuse/fuse.go (hunk 2 above)
Comment (nwt) [reply proc/fuse/fuse.go]: Done in a97847d99de10b5b91f5840cebcb2cb492349374.

File: zng/fuser/fuser.go (hunk 1 above)
Comment (nwt) [reply zng/fuser/fuser.go]: Done in 7f31208919e2c0013640c91752be292e36a5502f.

File: zng/fuser/fuser.go [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+	f.nbytes += len(rec.Raw)
+	if f.nbytes >= f.memMaxBytes {
+		var err error
+		f.spiller, err = spill.NewTempFile()
+		if err != nil {
+			return err
+		}
+		if err := f.spiller.Write(rec); err != nil {
+			return err
+		}
```
Comment (mattnibs) [inline zng/fuser/fuser.go:75]: Shouldn't this go after the recs iteration to keep the order or does this not matter?

Comment (nwt) [reply zng/fuser/fuser.go:75]: I think fuse should preserve order and created #1904 to track that.

**Comment by nwt** [comment]
`zng/fuser` didn't feel like the right home for `Fuser`, so I moved it back to `proc/fuse`.

**Review by alfred-landrum — APPROVED** [review]

**Pull request merged**